scripts/update_library_version.sh

```diff
@@ -0,0 +1,201 @@
+#!/bin/bash
+
+# Detailed help message
+print_help() {
+  echo "Usage: $0 <directory> <library_name> <new_version> <include_paths_with> <exclude_paths_with> <lock_files>"
+  echo
+  echo "Parameters:"
+  echo "  directory           The root directory to start searching for Pipfile and requirements-elyra.txt files."
+  echo "  library_name        The name of the library to update."
+  echo "  new_version         The new version to set for the library."
+  echo "  include_paths_with  A pipe-separated list of substrings; only files in directories containing at least one of these substrings will be processed."
+  echo "  exclude_paths_with  A pipe-separated list of substrings; files in directories containing any of these substrings will be excluded."
+  echo "  lock_files          Whether to run 'pipenv lock' after updating the library version (true or false)."
+  echo
+  echo "Examples:"
+  echo "  $0 ./myproject numpy 2.0.1 '' '' true"
+  echo "  $0 ./myproject pandas 2.2.2 'include|this' 'exclude|that' false"
+}
+
+# Check if the correct number of arguments are passed
+if [ "$#" -ne 6 ]; then
+  print_help
+  exit 1
+fi
+
+# Arguments
+directory=$1
+library_name=$2
+new_version=$3
+include_paths_with=$4
+exclude_paths_with=$5
+lock_files=$6
+
+# Print arguments
+echo "Arguments:"
+echo "  directory          = $directory"
+echo "  library_name       = $library_name"
+echo "  new_version        = $new_version"
+echo "  include_paths_with = $include_paths_with"
+echo "  exclude_paths_with = $exclude_paths_with"
+echo "  lock_files         = $lock_files"
+
+# Function to compare two versions
+# Returns 1 if the first version is greater, -1 if the second is greater, and 0 if they are equal
+version_compare() {
+  local ver1=$1
+  local ver2=$2
+
+  # Remove any non-numeric, non-dot characters from the beginning (before the first number, which represents the version specifier)
+  ver1=$(echo "$ver1" | sed 's/^[^0-9.]*//')
+  ver2=$(echo "$ver2" | sed 's/^[^0-9.]*//')
+
+  # Split each segment of the version numbers
+  IFS='.' read -r -a ver1_parts <<< "$ver1"
+  IFS='.' read -r -a ver2_parts <<< "$ver2"
+
+  for ((i = 0; i < ${#ver1_parts[@]} || i < ${#ver2_parts[@]}; i++)); do
+    # Use 0 if a part is missing
+    v1=${ver1_parts[i]:-0}
+    v2=${ver2_parts[i]:-0}
+    
+    # Compare the parts
+    if ((v1 > v2)); then
+      return 1
+    elif ((v1 < v2)); then
+      return -1
```

Comment:
This will effectively be 255 since you can't return negative values in bash IMO. So maybe let's use 255 directly to avoid possible confusion?

Reply:
@jstourac good catch, I changed the function using your suggestion.